cluster-autoscaler/Makefile

```diff
@@ -35,7 +35,6 @@ export DOCKER_CLI_EXPERIMENTAL := enabled
 build: build-arch-$(GOARCH)
 
 build-arch-%: clean-arch-%
-	$(ENVVAR) GOOS=$(GOOS) GOARCH=$* go build ${LDFLAGS_FLAG} ${TAGS_FLAG} ./...
```

Comment:
With this change `build-arch-%` is the same as `build-binary-arch-%`. Do we need both?

Reply:
good question, i don't _think_ so as they all seem to create the same binary target

Reply:
i looked into this a little more and i think we need to keep the build-arch target to satisfy the all target in https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/Makefile#L2

i have a feeling we will break a build _somewhere_ if we remove the build-arch target

Reply:
i think you are correct though, we could standardize around either build/build-arch or build-binary/build-binary-arch. i'll try out a few different combos.